### PR TITLE
Add inputs_commitment to TransactionEssence

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add inputs_commitment to TransactionEssence

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bee-message/src/payload/transaction/essence/regular.rs [hunk 1]
```diff
@@ -19,6 +19,7 @@ use alloc::vec::Vec;
 pub struct RegularTransactionEssenceBuilder {
     network_id: u64,
     inputs: Vec<Input>,
+    inputs_commitment: [u8; 32],
```
Comment (thibault-martinez) [inline bee-message/src/payload/transaction/essence/regular.rs:22]: It's before the inputs https://github.com/iotaledger/tips/pull/40/commits/1632afd81ddffe564a29f00bdd2585c1af8e3e39

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: bee-message/src/payload/transaction/essence/regular.rs (hunk 1 above)
Comment (Thoralf-M) [reply bee-message/src/payload/transaction/essence/regular.rs:22]: The tip will be changed :trollface:

Comment (Thoralf-M) [reply bee-message/src/payload/transaction/essence/regular.rs:22]: https://github.com/iotaledger/tips/pull/40/commits/574bcdcdc8a113d878a88daac3fccb03006d85ab

File: bee-message/src/payload/transaction/essence/regular.rs [hunk 2]
```diff
@@ -44,6 +45,12 @@ impl RegularTransactionEssenceBuilder {
         self
     }
 
+    /// Add inputs_commitment to a [`RegularTransactionEssenceBuilder`].
```
Comment (thibault-martinez) [inline bee-message/src/payload/transaction/essence/regular.rs]:
```suggestion
    /// Adds an inputs commitment to a [`RegularTransactionEssenceBuilder`].
```

**Review by thibault-martinez — APPROVED** [review]

**Review by bingyanglin — APPROVED** [review]

File: bee-message/src/payload/transaction/essence/regular.rs [hunk 3]
```diff
@@ -102,6 +110,8 @@ pub struct RegularTransactionEssence {
     #[packable(verify_with = verify_inputs)]
     #[packable(unpack_error_with = |e| e.unwrap_packable_or_else(|p| Error::InvalidInputCount(p.into())))]
     inputs: BoxedSlicePrefix<Input, InputCount>,
+    /// BLAKE2b-256 hash of the serialized outputs referenced in inputs by their OutputId.
+    inputs_commitment: [u8; 32],
```
Comment (grtlr) [inline bee-message/src/payload/transaction/essence/regular.rs:114]: I wonder if we eventually want to capture this information in a wrapper type. I wouldn't do it in this PR, but if you agree we could open an issue.

**Review by grtlr — APPROVED** [review]

File: bee-message/src/payload/transaction/essence/regular.rs (hunk 3 above)
Comment (thibault-martinez) [reply bee-message/src/payload/transaction/essence/regular.rs:114]: Yeah I would also do that. And same for token tag which is still an array. I'll merge now but pls open an issue 👍

**Pull request merged**